embrace-android-delivery/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/intake/IntakeService.kt

```diff
@@ -20,4 +20,9 @@ interface IntakeService : Shutdownable {
      * Stores a payload on disk as its JSON representation.
      */
     fun take(intake: Envelope<*>, metadata: StoredTelemetryMetadata)
+
+    /**
+     * Caches a payload on disk as its JSON representation.
+     */
+    fun cache(intake: Envelope<*>, metadata: StoredTelemetryMetadata)
```

Comment:
Do we need another method in the interface if we can tell whether a payload is complete to determine whether it's a cache or take scenario?

Reply:
Updated to just use one method.